### DISCUSSION TITLE
problem with number of processors used for simulations

### DISCUSSION THREAD

**Discussion by Oops-Qiao** [idaholab/moose#23482] [Q&A General] [answered]
Hello,

I have a quick question regarding the following code to run my simulations.
when I use  mpiexec -n 2  ~/projects/fly-opt -i test.i to run my test, terminal reports an error like below

However, when I use mpiexec  ~/projects/fly-opt -i test.i to run my test, it works well.

I want to use as many processors as I can to speed up the simulations. How to understand this? Thanks very much.

Best regards,
Jeery

**Comment by Oops-Qiao** [comment]
@GiudGiud Perhaps you have seen this before? :)

**Comment by GiudGiud** [comment, marked as answer]
Hello

This has been answered before on the forum indeed.
Preconditioning defaults are different in parallel and in serial. The serial one is more effective but does not scale to parallel. You ll need to work on getting the appropriate preconditioning. Some info here:
https://mooseframework.inl.gov/source/executioners/Steady.html

Please stop posting screenshots. It's better to copy paste text inside triple backquotes. 

Guillaume

**Reply by Oops-Qiao** [reply]
> Hello
> 
> This has been answered before on the forum indeed. Preconditioning defaults are different in parallel and in serial. The serial one is more effective but does not scale to parallel. You ll need to work on getting the appropriate preconditioning. Some info here: https://mooseframework.inl.gov/source/executioners/Steady.html
> 
> Please stop posting screenshots. It's better to copy paste text inside triple backquotes.
> 
> Guillaume

Thanks a lot Guillaume. I am sorry and the screenshot is deleted. No more next time.

Jerry
